kubernetes/cronjob.yaml

```
@@ -0,0 +1,47 @@
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  labels:
+    app.kubernetes.io/name: vault-snapshot
+    app.kubernetes.io/version: v0.1.0
+  name: vault-snapshot
+spec:
+  jobTemplate:
+    schedule: 0 4 * * *
+    metadata:
+      labels:
+        app.kubernetes.io/name: vault-snapshot
+        app.kubernetes.io/version: v0.1.0
+    spec:
+      template:
+        metadata:
+          labels:
+            app.kubernetes.io/name: vault-snapshot
+            app.kubernetes.io/version: v0.1.0
+        spec:
+          containers:
+          - name: vault-snapshot
+            env:
+            - name: S3_HOST
+              value: s3.example.com
+            - name: S3_BUCKET
+              value: bucketname
+            - name: S3_URI
+              value: s3://bucketname
+            - name: VAULT_ROLE
+              value: vault-snapshot
+            - name: VAULT_ADDR
+              value: https://vault.example.com
+            - name: AWS_SECRET_ACCESS_KEY
+              valueFrom:
+                secretKeyRef:
+                  key: aws_secret_access_key
+                  name: vault-snapshot-credentials
+            - name: AWS_ACCESS_KEY_ID
+              valueFrom:
+                secretKeyRef:
+                  key: aws_access_key_id
+                  name: vault-snapshot-credentials
+              image: ghcr.io/adfinis/vault-snapshot:0.1.-0
+            imagePullPolicy: IfNotPresent
```

Again an extension regarding restricted environments. We might run into a situation where the container filesystem is mounted read-only and we can't even write temporary files, so we should request and mount an `emptyDir`:
```suggestion
            imagePullPolicy: IfNotPresent
            volumes:
            - name: backup-dir
              emptyDir: {}
```